src/ws/session.c

```diff
@@ -630,9 +657,19 @@ main (int argc,
   int status;
   int res;
   int i;
+  struct rlimit fileno_limit;
 
   program_name = basename (argv[0]);
 
+  /* we don't expect cockpit-session to use more than a handful of fds, so that we can reliably
+   * close all but the ones that we expect for the forked cockpit-bridge. */
+  if (getrlimit (RLIMIT_NOFILE, &orig_fileno_limit) < 0)
+    errx (1, "failed to get RLIMIT_NOFILE");
+  fileno_limit.rlim_cur = SESSION_MAX_FDS;
+  fileno_limit.rlim_max = orig_fileno_limit.rlim_max; /* we need to be able to restore it once unprivileged */
+  if (setrlimit (RLIMIT_NOFILE, &fileno_limit) < 0)
```

Comment:
Lowering the limits does not actually close any file descriptor, neither immediately in the current process, nor across fork or exec, right?

Reply:
Correct. However, @allisonkarlitskaya already objected that this will interact badly with `pam_limits`, so we'll discard that approach anyway.